viadot/sources/vid_club.py

```diff
@@ -80,70 +77,140 @@ def build_query(
-            )
         if to_date == None:
-            datetime.today().strftime("%Y-%m-%d")
+            end_date = datetime.today().date()
+        else:
+            end_date = datetime.strptime(to_date, "%Y-%m-%d").date()
+        start_date = datetime.strptime(from_date, "%Y-%m-%d").date()
 
         from_date_obj = datetime.strptime(from_date, "%Y-%m-%d")
         oldest_date_obj = datetime.strptime("2022-03-22", "%Y-%m-%d")
         delta = from_date_obj - oldest_date_obj
 
         if delta.days < 0:
-            raise ValidationError("from_date cannot be earlier than 2023-03-22!!")
+            raise ValidationError("from_date cannot be earlier than 2022-03-22.")
 
         to_date_obj = datetime.strptime(to_date, "%Y-%m-%d")
         delta = to_date_obj - from_date_obj
 
         if delta.days < 0:
-            raise ValidationError("to_date cannot be earlier than from_date!")
+            raise ValidationError("to_date cannot be earlier than from_date.")
+        interval = timedelta(days=days_interval)
+        starts = []
+        ends = []
+
+        period_start = start_date
+        while period_start < end_date:
+            period_end = min(period_start + interval, end_date)
+            starts.append(period_start.strftime("%Y-%m-%d"))
+            ends.append(period_end.strftime("%Y-%m-%d"))
+            period_start = period_end
+        if len(starts) == 0 and len(ends) == 0:
+            starts.append(from_date)
+            ends.append(to_date)
+        return starts, ends
+
+    def check_connection(
+        self,
+        source: Literal["jobs", "product", "company", "survey"] = None,
+        from_date: str = "2022-03-22",
+        to_date: str = None,
+        items_per_page: int = 100,
+        region: str = "all",
+        url: str = None,
+    ) -> Tuple[Dict[str, Any], str]:
+        """
+        Initiate first connection to API to retrieve piece of data with information about type of pagination in API URL.
+        This option is added because type of pagination for endpoints is being changed in the future from page number to 'next' id.
+
+        Args:
+            source (Literal["jobs", "product", "company", "survey"], optional): The endpoint source to be accessed. Defaults to None.
+            from_date (str, optional): Start date for the query, by default is the oldest date in the data 2022-03-22.
+            to_date (str, optional): End date for the query. By default, datetime.today() will be used.
+            items_per_page (int, optional): Number of entries per page. 100 entries by default.
+            region (str, optional): Region filter for the query. Valid inputs: ["bg", "hu", "hr", "pl", "ro", "si", "all"]. Defaults to "all".
+            url (str, optional): Generic part of the URL to Vid Club API. Defaults to None.
+
+        Returns:
+            Dict[str, Any], str: First response from API with JSON containing data and used URL string
```

Comment:
Add a period at the end of the sentence and double-check if the `return type` is correct.